Document httpapi and cliconf plugins (#49503)
ansible/ansible: docs/docsite/rst/plugins/cliconf.rst

.. _cliconf_plugins:

Cliconf Plugins
===============

.. contents::
   :local:
   :depth: 2

> Cliconf plugins are abstactions over the cli interface to network devices. They provide a standard interface

Reviewer: ```suggestion
Cliconf plugins are abstactions over the CLI interface to network devices. They provide a standard interface
```

> for Ansible to execute tasks on them.

Reviewer: ```suggestion
for Ansible to execute tasks on those network devices.
```

These plugins generally correspond one-to-one to network device platforms. The appropriate cliconf plugin will

> thus be automatically loaded from the network_os variable.

Reviewer: ```suggestion
thus be automatically loaded based on the ``network_os`` variable.
```

.. _enabling_cliconf:

Adding cliconf plugins
-------------------------

You can extend Ansible to support other network devices by dropping a custom plugin into the ``cliconf_plugins`` directory.

.. _using_cliconf:

Using cliconf plugins
------------------------

> The cliconf plugin to use is determined automatically from the ansible_network_os variable. There should be no reason to override this functionality

Reviewer: ```suggestion
The cliconf plugin to use is determined automatically from the ``ansible_network_os`` variable. There should be no reason to override this functionality.
```

> Most cliconf plugins can operate without configuration. A few have additional options that can be set to imact how

Reviewer: ```suggestion
Most cliconf plugins can operate without configuration. A few have additional options that can be set to impact how
```

> tasks are translated into cli commands.

Reviewer: ```suggestion
tasks are translated into CLI commands.
```

Plugins are self-documenting. Each plugin should document its configuration options.

.. _cliconf_plugin_list:

> Plugin List

Reviewer: ```suggestion
Plugin list
```

Author: Pretty sure this should still be title case. In any case, this is how it is in all other pages

Reviewer: not a biggie. We're slowly moving to sentence case so I comment as I notice it in reviews...
